app/models/hackney/repair_request.rb

```diff
@@ -83,27 +83,35 @@ def add_nested_errors
       case key
       when /^contact\.(.*)$/i
         contact.errors.add($1, msg)
-      when /^work_orders\[0\]\.(.*)$/i
-        work_orders[0].errors.add($1, msg)
+      when /^work_orders\[(\d+)\]\.(.*)$/i
+        work_orders[$1.to_i].errors.add($2, msg)
       end
     end
   end
 
   # TODO: API should give better error descriptions
   def self.errors_from_api(response)
-    response.map {|x| x["userMessage"] }.group_by do |x|
-      case x
-      when /Telephone/i
-        "contact.telephone_number"
-      when /Contact Name/i
-        "contact.name"
-      when /Problem/i
-        "description"
-      when /sorCode/i
-        "work_orders[0].sor_code"
-      else
-        "base"
-      end
+    err = {}
+    response.each do |x|
+      key = parse_json_pointer(x["source"])
+      err[key] ||= []
```

Comment:
Does `err[key]` returns `nil` or an empty `array`?

Reply:
After that line it should be an empty array.

Reply:
If always should be an empty array you may as well then just assign it instead of conditional assignment :)

Reply:
Ohh I see. The point of doing this is that `key` may appear twice, and if we don't test with `||=`, second error will replace the first instead of getting appended on the next line.